<?xml version="1.0" encoding="UTF-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from ~/ros2_ws/install/pathfinder/share/pathfinder/description/robot.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="pathfinder">
   <!-- Standard inertial calculations https://en.wikipedia.org/wiki/List_of_moments_of_inertia -->
   <!--materials-->
   <material name="white">
      <color rgba="1 1 1 1" />
   </material>
   <material name="black">
      <color rgba="0 0 0 1" />
   </material>
   <material name="red">
      <color rgba="1 0 0 1" />
   </material>
   <material name="fontys_purple">
      <color rgba="0.4 0.2 0.4 1" />
   </material>
   <!-- BASE LINK -->
   <link name="base_link" />
   <!-- CHASSIS LINK -->
   <joint name="chassis_joint" type="fixed">
      <parent link="base_link" />
      <child link="chassis" />
      <origin xyz="-0.1 0 0" />
   </joint>
   <link name="chassis">
      <visual>
         <origin rpy="0 0 1.5707963267948966" xyz="0.15 0 0" />
         <geometry>
            <mesh filename="file:///home/hcdiekmann/ros2_ws/install/pathfinder/share/pathfinder/description/meshes/pathfinder_chassis.stl" scale="0.001 0.001 0.001" />
            <!-- scale is in meters -->
         </geometry>
         <material name="fontys_purple" />
      </visual>
      <collision>
         <origin xyz="0.15 0 0.075" />
         <geometry>
            <box size="0.3 0.3 0.15" />
         </geometry>
      </collision>
      <inertial>
         <origin rpy="0 0 0" xyz="0.15 0 0.075" />
         <mass value="0.5" />
         <inertia ixx="0.004687499999999999" ixy="0.0" ixz="0.0" iyy="0.004687499999999999" iyz="0.0" izz="0.0075" />
      </inertial>
   </link>
   <!-- LEFT WHEEL LINK -->
   <joint name="left_wheel_joint" type="continuous">
      <parent link="base_link" />
      <child link="left_wheel" />
      <origin rpy="-1.5707963267948966 0 0" xyz="0 0.175 0" />
      <axis xyz="0 0 1" />
   </joint>
   <link name="left_wheel">
      <visual>
         <geometry>
            <cylinder length="0.04" radius="0.05" />
         </geometry>
         <material name="black" />
      </visual>
      <collision>
         <geometry>
            <cylinder length="0.04" radius="0.05" />
         </geometry>
      </collision>
      <inertial>
         <origin rpy="0 0 0" xyz="0 0 0" />
         <mass value="0.1" />
         <inertia ixx="7.583333333333335e-05" ixy="0.0" ixz="0.0" iyy="7.583333333333335e-05" iyz="0.0" izz="0.00012500000000000003" />
      </inertial>
   </link>
   <gazebo reference="left_wheel">
      <material>Gazebo/Black</material>
   </gazebo>
   <!-- RIGHT WHEEL LINK -->
   <joint name="right_wheel_joint" type="continuous">
      <parent link="base_link" />
      <child link="right_wheel" />
      <origin rpy="1.5707963267948966 0 0" xyz="0 -0.175 0" />
      <axis xyz="0 0 -1" />
   </joint>
   <link name="right_wheel">
      <visual>
         <geometry>
            <cylinder length="0.04" radius="0.05" />
         </geometry>
         <material name="black" />
      </visual>
      <collision>
         <geometry>
            <cylinder length="0.04" radius="0.05" />
         </geometry>
      </collision>
      <inertial>
         <origin rpy="0 0 0" xyz="0 0 0" />
         <mass value="0.1" />
         <inertia ixx="7.583333333333335e-05" ixy="0.0" ixz="0.0" iyy="7.583333333333335e-05" iyz="0.0" izz="0.00012500000000000003" />
      </inertial>
   </link>
   <gazebo reference="right_wheel">
      <material>Gazebo/Black</material>
   </gazebo>
   <!-- CASTER WHEEL LINK -->
   <joint name="caster_wheel_joint" type="fixed">
      <parent link="chassis" />
      <child link="caster_wheel" />
      <origin xyz="0.24 0 0" />
   </joint>
   <link name="caster_wheel">
      <visual>
         <geometry>
            <sphere radius="0.05" />
         </geometry>
         <material name="black" />
      </visual>
      <collision>
         <geometry>
            <sphere radius="0.05" />
         </geometry>
      </collision>
      <inertial>
         <origin rpy="0 0 0" xyz="0 0 0" />
         <mass value="0.1" />
         <inertia ixx="0.00010000000000000005" ixy="0.0" ixz="0.0" iyy="0.00010000000000000005" iyz="0.0" izz="0.00010000000000000005" />
      </inertial>
   </link>
   <gazebo reference="caster_wheel">
      <material>Gazebo/Black</material>
      <mu1 value="0.001" />
      <!-- friction -->
      <mu2 value="0.001" />
   </gazebo>
   <gazebo>
      <plugin filename="libgazebo_ros_diff_drive.so" name="diff_drive">
         <ros>
            <!--<namespace>-->
            <!--<remapping tf>-->
         </ros>
         <!-- Wheel Information -->
         <left_joint>left_wheel_joint</left_joint>
         <right_joint>right_wheel_joint</right_joint>
         <wheel_separation>0.35</wheel_separation>
         <wheel_diameter>0.1</wheel_diameter>
         <!-- Limits -->
         <max_wheel_torque>180</max_wheel_torque>
         <max_wheel_acceleration>8.0</max_wheel_acceleration>
         <!-- Output -->
         <odometry_frame>odom</odometry_frame>
         <robot_base_frame>base_link</robot_base_frame>
         <publish_odom>true</publish_odom>
         <publish_odom_tf>true</publish_odom_tf>
         <publish_wheel_tf>true</publish_wheel_tf>
      </plugin>
   </gazebo>
   <joint name="lidar_joint" type="fixed">
      <parent link="chassis" />
      <child link="laser_frame" />
      <origin rpy="0 0 0" xyz="0.1 0 0.175" />
   </joint>
   <link name="laser_frame">
      <visual>
         <geometry>
            <cylinder length="0.04" radius="0.05" />
         </geometry>
         <material name="red" />
      </visual>
      <collision>
         <geometry>
            <cylinder length="0.04" radius="0.05" />
         </geometry>
      </collision>
      <inertial>
         <origin rpy="0 0 0" xyz="0 0 0" />
         <mass value="0.1" />
         <inertia ixx="7.583333333333335e-05" ixy="0.0" ixz="0.0" iyy="7.583333333333335e-05" iyz="0.0" izz="0.00012500000000000003" />
      </inertial>
   </link>
   <gazebo reference="laser_frame">
      <material>Gazebo/Red</material>
      <sensor name="laser" type="ray">
         <pose>0 0 0 0 0 0</pose>
         <visualize>false</visualize>
         <update_rate>10</update_rate>
         <ray>
            <scan>
               <horizontal>
                  <samples>360</samples>
                  <min_angle>-3.14</min_angle>
                  <max_angle>3.14</max_angle>
               </horizontal>
            </scan>
            <range>
               <min>0.3</min>
               <max>10</max>
            </range>
         </ray>
         <plugin filename="libgazebo_ros_ray_sensor.so" name="laser_controller">
            <ros>
               <argument>~/out:=scan</argument>
            </ros>
            <output_type>sensor_msgs/LaserScan</output_type>
            <frame_name>laser_frame</frame_name>
         </plugin>
      </sensor>
   </gazebo>
</robot>